features/mbedtls/targets/TARGET_NUVOTON/TARGET_M480/ecp/ecp_internal_alt.c

```diff
@@ -589,7 +639,18 @@ NU_STATIC int internal_run_eccop(const mbedtls_ecp_group *grp,
         ret = MBEDTLS_ERR_ECP_BAD_INPUT_DATA;
         goto cleanup;
     }
-    
+
+    /* Open ECC accelerator
+     *
+     * To guarantee ECC accelerator operation is atomic, ECC accelerator open
+     * period must include:
+     * 1. Configure big-num parameters through internal_mpi_write_eccreg()
+     * 2. Trigger and wait
+     * 3. Read back result through internal_mpi_read_eccreg()
+     */
+    internal_open_ecc_ac();
+    ecc_ac_open = true;
```

Comment:
Can't you just put the `close` ahead of the `cleanup` label?

Reply:
@kjbracey-arm I must guarantee `internal_open_ecc_ac` and `internal_close_ecc_ac` are paired. When `MBEDTLS_MPI_CHK` evaluates false, it will go to `cleanup` label. So I can only call `internal_close_ecc_ac` after `cleanup` label for code concision.

```
#define MBEDTLS_MPI_CHK(f) do { if( ( ret = f ) != 0 ) goto cleanup; } while( 0 )
```

Reply:
Ah, missed that. Okay, that could be dealt with with another `cleanup_hw` label ahead of the close.

(Or, as you're in C++ now you start putting in RAII stuff that automatically destroys the things)

But the bool is fine.

Reply:
(Except I still think you could continue to make these calls in the `init`/`free` anyway, as per other comments).

Reply:
@kjbracey-arm As above, ecc AC init/free are moved back to `mbedtls_internal_ecp_init` / `mbedtls_internal_ecp_free`.